gobblin-utility/src/main/java/gobblin/util/FileListUtils.java

```diff
@@ -54,6 +55,46 @@ public boolean accept(Path path) {
   }
 
   /**
+   * Method to list out all files, or directory if no file exists, under a specified path.
+   */
+  public static List<FileStatus> listMostNestedPathRecursively(FileSystem fs, Path path) throws FileNotFoundException,
+      IOException {
+    return listMostNestedPathRecursively(fs, path, new PathFilter() {
```

Comment:
It's worth defining this special `PathFilter` as a constant like `private static final PathFilter NO_OP_PATH_FILTER = ...`.